src/pymor/models/interface.py

```diff
@@ -30,6 +30,9 @@ class Model(CacheableObject, ParametricObject):
         `True` if the model describes a linear problem.
     products
         Dict of inner product operators associated with the model.
+    order
+        The number of equations defining the |Model|, usually the dimension of
```

Comment:
Any reason for 'usually'? I would have hoped this to be the formal definition of `order` ...

Reply:
I think the counterexample are second-order systems, where the `solution_space` would be the block space of positions and velocities, but it would be more natural for `order` to refer to the number of equations.

Reply:
I thought we specifically chose `solution_space` over `state_space`, because it does not have have the velocities in it? (Regarding letting `SecondOrderModel` inherit from a first order model, as recently discussed, this might be an argument for `to_first_order` instead.) Anyway, if `order` has no properly defined meaning for a general `Model`, then we should not add it to the interface. But I'd still hope we can drop the 'usually' ...

Reply:
> I thought we specifically chose `solution_space` over `state_space`, because it does not have have the velocities in it?

I remember that we discussed this, but I think now that `solution_space` of `SecondOrderModels` should also contain velocities. For instance, the output of a `SecondOrderModel` can depend on the velocities. (The situation with `LinearDelayModel` becomes more complicated if the output can also have delays.)

> Anyway, if `order` has no properly defined meaning for a general `Model`, then we should not add it to the interface. But I'd still hope we can drop the 'usually' ...

If we agree that the definition of order as the number of equations is ok, then `order` could be `op.range.dim`, where `op` is one of the operators in the state equation.

Reply:
BTW, you could also always add a switch like `return_veclocities` to `solve` to get another array with the velocity vectors ..

Reply:
How about the following: We remove the 'usually' for now. As soon as we have a situation where the definition is not appropriate anymore, we reword it to be more general. (I still think that `SecondOrderModel` should not be a problem.)

Reply:
Sorry, I guess we missunderstood each other: I was thinking about `Dimension of the solution_space` as the definition/docstring for `order`. If you still don't like it, we can also leave it as is for now and discuss next year.

Reply:
Yes, I guess "Dimension of the `solution_space`" makes more sense based on the default implementation of `order`.